### PR TITLE
[bitnami/nats] Store password in secrets

### PR DESCRIPTION
**Pull request by nobiit**
### Description of the change
Store password & token for Nats as secret standalone

### Benefits
It can be reused via environment variable in other apps

### Possible drawbacks
None

### Applicable issues
None

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/nats/templates/secrets.yaml [hunk 1]
```diff
@@ -11,7 +11,107 @@ metadata:
...
+  token: {{ .Values.auth.token | quote }}
+  {{- end }}
+  {{- end }}
+  {{- if .Values.cluster.auth.enabled }}
+  {{- if .Values.cluster.auth.user }}
+  cluster-password: {{ $clusterAuthPwd | quote }}
+  {{- else if .Values.cluster.auth.token }}
+  token: {{ .Values.cluster.auth.token | quote }}
+  {{- end }}
+  {{- end }}
```
Comment (migruiz4) [inline bitnami/nats/templates/secrets.yaml:34]: The key `token` is duplicated. I would like to suggest the following unique keys: 
- `password`
- `token`
- `cluster-password`
- `cluster-token`

Comment (migruiz4) [reply bitnami/nats/templates/secrets.yaml:34]: In addition, please move these conditionals out of the `{{ if .Values.configuration }}` conditional.

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Thank you for your contribution @nobiit.
Please take a look at my suggestions.

In addition, the Nats statefulset uses the following volume 
```
        - name: config
          secret:
            secretName: {{ include "nats.secretName" . }}
```
To avoid mounting the auth related keys as files, please add the `items`:
```
        - name: config
          secret:
            secretName: {{ include "nats.secretName" . }}
            items:
              - key: {{ .Values.natsFilename }}.conf
                path: {{ .Values.natsFilename }}.conf
```

**Comment by migruiz4** [comment]
Hi @nobiit,
I have two more requests before we can merge this PR:
- Please resolve the conflicts.
- The Github action "Update README metadata" is failing because contributors do not have permission to push to your repository. You need to either run the [helm-readme-generator tool](https://github.com/bitnami-labs/readme-generator-for-helm) locally or allow contributions so the bitnami-bot can push the updated README.

**Comment by nobiit** [comment]
Hey @migruiz4 , I will try to update this PR as soon as possible. Thank you for your comments

**Comment by nobiit** [comment]
Hey @migruiz4 

> * The Github action "Update README metadata" is failing because contributors do not have permission to push to your repository. You need to either run the [helm-readme-generator tool](https://github.com/bitnami-labs/readme-generator-for-helm) locally or allow contributions so the bitnami-bot can push the updated README.


I sent an invite to allow bitnami-bot to contribute to nobidev/bitnami-charts, how do I get bitnami-bot to accept it? I'm having multiple PRs that fail when I run readme-generator-for-helm and it could confuse the Bitnami collaborators as incomplete PRs

**Comment by migruiz4** [comment]
> I sent an invite to allow bitnami-bot to contribute to nobidev/bitnami-charts, how do I get bitnami-bot to accept it?

I'm sorry but we can't get the bitnami bot to accept any invitation, you may need to allow all contributors to make changes in your bitnami/charts fork or run the tool yourself.

> I'm having multiple PRs that fail when I run readme-generator-for-helm and it could confuse the Bitnami collaborators as incomplete PRs

What issues are you experiencing when running the tool? I may be able to help you.

**Comment by nobiit** [comment]
Hey @migruiz4 , I have updated the README.md, can you review the PR again ?

**Comment by migruiz4** [comment]
Hi @nobiit, sorry for the late response, could you please resolve the conflicts?

**Review by migruiz4 — CHANGES_REQUESTED** [review]
I have noticed that previously requested changes haven't been applied.

**Pull request closed without merging**